src/runtime/tests/particle-interface-loading-test.ts

```diff
@@ -293,4 +293,188 @@ describe('particle interface loading', () => {
     const fooHandle2 = await handleNGFor('crdt-key', varStorageProxy2, arc2.idGenerator, null, true, true, 'varHandle') as SingletonHandle<Entity>;
     assert.deepStrictEqual(await fooHandle2.fetch(), new fooClass({value: 'Not created!'}));
   });
+
+  it('onReady sees overriden values in onCreate', async () => {
+    const manifest = await Manifest.parse(`
+      schema Foo
+        value: Text
+
+      particle UpdatingParticle in 'updating-particle.js'
+        bar: reads writes Foo
+      recipe
+        h1: use *
+        UpdatingParticle
+          bar: h1
+    `);
+    assert.lengthOf(manifest.recipes, 1);
+    const recipe = manifest.recipes[0];
+    const loader = new Loader(null, {
+      'updating-particle.js': `
+        'use strict';
+        defineParticle(({Particle}) => {
+          var handlesSynced = 0;
+          return class extends Particle {
+            async onCreate() {
```

Comment:
You're doing an interesting thing here. onCreate is not an async function in the base class. You are making it async and awaiting on the `barHandle.set`, but the base class will not wait... to sum up, you should not do that. The setting of bar may still be in progress when onReady() is called.

Open questions: Should onCreate / onReady etc. be async / suspend functions? I think it may be useful, given that they should be used for setting initial state. Shall we discuss this during the TEA time?

Reply:
I think this is all fixed up now... but wouldn't mind a second set of eyes to verify it matches what we discussed in Tea.